src/Hamiltonians/reduced_density_matrix.jl

```diff
@@ -243,3 +243,103 @@ function (calc!::ReducedDensityMatrixCalculcator!{TT, P})(result, pair) where {T
     end
     return result
 end
+
+"""
+     TestOneParticleDensity(v; normalize=true) <: AbstractOperator{typeof(v)}
+A one particle operator constructed from a provided test vector `v`. An expectation value
+with this operator yields a lower bound on the largest eigenvalue (and an upper bound on the
+smallest eigenvalue) of the one-particle density matrix.
+If `normalize` is true (default), the vector is normalized before use.
+
+```math
+    ρ̂ = ∑_{ij} v_i^* v_j â^†_{i} â_{j}
+```
+"""
+struct TestOneParticleDensity{T,V<:SVector{<:Any,T},M} <: AbstractObservable{T}
```

Comment:
```suggestion
struct TestOneParticleDensity{T,V<:Vector{T},M} <: AbstractObservable{T}
```
If this is going to be (potentially) useful later for evaluating a lower bound to the eigenvalue of the 1RDM in a scenario where the vector is updated all the time, then the test vector should rather be stored in a mutable container (i.e. `Vector` rather than `SVector`).

Reply:
I still feel that wrapping a vector that can be modified might be faster when doing this often and with larger basis sets than constructing a new `SVector` each time.